library/std/src/fs.rs

```diff
@@ -615,6 +615,223 @@ impl File {
         self.inner.datasync()
     }
 
+    /// Acquire an exclusive advisory lock on the file. Blocks until the lock can be acquired.
+    ///
+    /// This acquires an exclusive advisory lock; no other file handle to this file may acquire
+    /// another lock.
+    ///
+    /// If this file handle, or a clone of it, already holds an advisory lock the exact behavior is
+    /// unspecified and platform dependent, including the possibility that it will deadlock.
+    /// However, if this method returns, then an exclusive lock is held.
+    ///
+    /// If the file not open for writing, it is unspecified whether this function returns an error.
+    ///
+    /// Note, this is an advisory lock meant to interact with [`lock_shared`], [`try_lock`],
+    /// [`try_lock_shared`], and [`unlock`]. Its interactions with other methods, such as [`read`]
+    /// and [`write`] are platform specific, and it may or may not cause non-lockholders to block.
+    ///
+    /// # Platform-specific behavior
+    ///
+    /// This function currently corresponds to the `flock` function on Unix with the `LOCK_EX` flag,
+    /// and the `LockFileEx` function on Windows with the `LOCKFILE_EXCLUSIVE_LOCK` flag. Note that,
+    /// this [may change in the future][changes].
```

Comment:
It must not change for programs compiled with different rustc versions to interoperate wrt locking, right?

Reply:
I included `may change in the future` because that language is used in all the other places that `Platform-specific behavior` is documented, that I could find. Is there guidance on when to include or not include it?

It seems appropriate to include it to me, because the implementation can be changed and still interoperate. For example, `LockFile` can be called instead of `LockFileEx` in the case of `lock()` and `lock_shared()`. Similarly, you can imagine that a new syscall might be added to Unix and/or Windows which can be used as an alternative.